### PR TITLE
style(suite): Redesign onboarding pages

### LINKED ISSUE #10575: Update Onboarding pages
Screenshot 2024-01-09 at 14 46 31

Screenshot 2024-01-09 at 14 47 09

Screenshot 2024-01-09 at 14 48 00

Screenshot 2024-01-09 at 14 49 16

### PR DESCRIPTION
**Pull request by jvaclavik**
## Related Issue

Resolve #10575

## Screenshots:

desktop:
image

mobile:

image

image

### CODE REVIEW

File: packages/suite/src/views/onboarding/steps/DeviceTutorial.tsx [hunk 1]
```diff
@@ -75,23 +69,35 @@ export const DeviceTutorial = () => {
...
-                            >
-                                <Translation id="TR_SKIP" />
-                            </ActionButton>
-                        )}
+                        <ButtonContainer>
+                            {isActionAbortable && (
+                                <Button
+                                    variant="tertiary"
+                                    size="tiny"
+                                    onClick={() =>
```
Comment (peter-sanderson) [inline packages/suite/src/views/onboarding/steps/DeviceTutorial.tsx]: extract out of render

**Review by peter-sanderson — APPROVED** [review]

**Comment by AdamSchinzel** [comment]
Existing issue I think that we can solve, it looks weird that it is visible between the navigation and the top edge of the screen.

94C9D516-3EB5-4FA9-8552-B0AEFE475F50

File: packages/suite-data/files/translations/cs.json [hunk 2]
```diff
@@ -1719,7 +1719,7 @@
   "TR_TREZOR_BRIDGE_IS_NOT_RUNNING": "Trezor Bridge není spuštěn",
   "TR_TREZOR_DEVICE_TUTORIAL_CANCELED_HEADING": "Tutorial zrušen",
   "TR_TREZOR_DEVICE_TUTORIAL_COMPLETED_HEADING": "Tutorial dokončen",
-  "TR_TREZOR_DEVICE_TUTORIAL_DESCRIPTION": "Naučte se používat zařízení pomocí krátkého tutorialu",
```
Comment (AdamSchinzel) [inline packages/suite-data/files/translations/cs.json]: Should be updated in Crowdin not in the code

Comment (AdamSchinzel) [reply packages/suite-data/files/translations/cs.json]: I have just updated it in Crowdin

File: packages/suite/src/components/onboarding/SkipStepConfirmation.tsx
Comment (dahaca) [inline packages/suite/src/components/onboarding/SkipStepConfirmation.tsx:1]: Let's start using the `DialogModal` component with this one! ;D

File: packages/suite/src/components/onboarding/OnboardingProgressBar.tsx [hunk 3]
```diff
@@ -74,15 +73,15 @@ const Label = styled.div`
...
+    border-bottom: 1px solid ${({ theme }) => theme.borderOnElevation0};
+    margin: ${spacingsPx.md} ${spacingsPx.lg};
 
     @media (max-width: ${variables.SCREEN_SIZE.XL}) {
         margin: 15px;
     }
 
     @media (max-width: ${variables.SCREEN_SIZE.SM}) {
-        margin: 10px;
+        margin: 15px 10px;
```
Comment (dahaca) [inline packages/suite/src/components/onboarding/OnboardingProgressBar.tsx]: nit: use spacings

File: packages/suite/src/components/onboarding/OnboardingProgressBar.tsx [hunk 4]
```diff
@@ -47,15 +47,15 @@ const IconWrapper = styled.div<{ stepCompleted?: boolean; active?: boolean }>`
     ${props =>
         props.stepCompleted &&
         css`
-            background: ${({ theme }) => theme.BG_LIGHT_GREY};
+            background: ${({ theme }) => theme.transparent};
```
Comment (dahaca) [inline packages/suite/src/components/onboarding/OnboardingProgressBar.tsx]: nit: no `theme` needed

File: packages/suite/src/components/onboarding/SkipStepConfirmation.tsx
Comment (jvaclavik) [reply packages/suite/src/components/onboarding/SkipStepConfirmation.tsx:1]: Yes we should replace it. But I suppose that we are missing an overlay + portal, right? I checked implementation of Modal and I didn't find it there

Screenshot 2024-01-12 at 8 51 27

File: packages/suite/src/views/onboarding/steps/DeviceTutorial.tsx (hunk 1 above)
Comment (jvaclavik) [reply packages/suite/src/views/onboarding/steps/DeviceTutorial.tsx]: ee7032b27950ae4712990440f21d6ee7b156973e

**Comment by jvaclavik** [comment]
> Existing issue I think that we can solve, it looks weird that it is visible between the navigation and the top edge of the screen.
> 
> 94C9D516-3EB5-4FA9-8552-B0AEFE475F50

ee7032b27950ae4712990440f21d6ee7b156973e

File: packages/suite/src/components/onboarding/OnboardingProgressBar.tsx (hunk 4 above)
Comment (jvaclavik) [reply packages/suite/src/components/onboarding/OnboardingProgressBar.tsx]: ee7032b27950ae4712990440f21d6ee7b156973e

File: packages/suite/src/components/onboarding/OnboardingProgressBar.tsx (hunk 3 above)
Comment (jvaclavik) [reply packages/suite/src/components/onboarding/OnboardingProgressBar.tsx]: ee7032b27950ae4712990440f21d6ee7b156973e

**Comment by AdamSchinzel** [comment]
Unchecked checkboxes look not really good to me in dark mode
077AD6F5-3FF1-44AC-9C14-5C0636EC0FF8

File: packages/suite/src/components/onboarding/SkipStepConfirmation.tsx
Comment (dahaca) [reply packages/suite/src/components/onboarding/SkipStepConfirmation.tsx:1]: Ah you are right, I should have elaborated!

`Modal` and `DialogModal` from the `components` package are raw modal components, meaning there is no modal logic in there. It is handled in Suite itself, in `components/suite/modals`. 

And even there it's not so simple!

There is the [`Modal`](https://github.com/trezor/trezor-suite/blob/19a03812380dd926bad21bee3c2b58a105e6ba72/packages/suite/src/components/suite/modals/Modal/Modal.tsx#L9) component, which serves as a wrapper for the shared [`Modal`](https://github.com/trezor/trezor-suite/blob/975e015d9290a6d23702d6d72207859e5245067c/packages/components/src/components/modals/Modal/Modal.tsx#L189), with all the Suite-specific logic included. As you can see, it has `renderer: View = DefaultRenderer`, and the `DefaultRenderer` component has all the portal stuff and so on. 

Then take a look at the [`DevicePromptModal`](https://github.com/trezor/trezor-suite/blob/4603725193ae07219e340bde749f0164c585ab56/packages/suite/src/components/suite/modals/ReduxModal/DeviceContextModal/DevicePromptModal.tsx#L91): it uses  the **suite** `Modal` but passes a different **renderer**! What is does, is providing a `Modal` with a pre-defined interface and an API different from the default `Modal`. It technically could have been in the shared components as well but at the time we did not even have a unified device prompt modal at all, so it was the first step :) 

To sum it all up, here is a working suite-adapted `DialogModal`! And I would place it with the `DefaultRenderer` and `Modal`. 

```
export const DialogRenderer = ({
    headerComponent,
    isCancelable,
    onCancel,
    ...rest
}: ModalProps): ReactPortal | null => {
    const modalTarget = useModalTarget();

    if (!modalTarget) return null;

    const modal = (
        <ModalEnvironment onClickBackdrop={isCancelable ? onCancel : undefined}>
            <BaseDialogModal isCancelable={isCancelable} onCancel={onCancel} {...rest} />
        </ModalEnvironment>
    );

    return createPortal(modal, modalTarget);
};

export const DialogModal = (props: DialogModalProps) => (
    <Modal {...props} renderer={DialogRenderer} />
);

```

_cc: @AdamSchinzel @peter-sanderson maybe would be useful for you guys as well!_

**Comment by dahaca** [comment]
> Unchecked checkboxes look not really good to me in dark mode

5f1006bc49

An example of using `elevation` 🙏

File: packages/suite/src/components/onboarding/SkipStepConfirmation.tsx
Comment (jvaclavik) [reply packages/suite/src/components/onboarding/SkipStepConfirmation.tsx:1]: Nice! Thank you 🙏

**Comment by jvaclavik** [comment]
All issues should be fixed @AdamSchinzel @dahaca

**Review by dahaca — APPROVED** [review]
✨

**Review by AdamSchinzel — APPROVED** [review]
lgtm

**Pull request merged**